src/components/ExternalLink/ExternalLink.tsx

```diff
@@ -15,6 +16,10 @@ export interface ExternalLinkProps extends LinkProps {
    * Adds the nofollow annotation to the anchor's rel attribute
    */
   noFollow?: boolean;
+  /**
+   * Show a custom icon next to the link. `hasIcon` must be false
+   */
+  rightIcon?: ReactNode;
```

Comment:
Why not allow this prop to override hasIcon?

```tsx
const hasIcon = props.hasIcon || Boolean(rightIcon);
const icon =  hasIcon ? rightIcon ?? <ExternalLinkIcon /> : null;

...

{children} {icon}
```

Reply:
Whoops, yeah that makes way more sense 👍 Updated.